index.js

```diff
@@ -84,8 +84,24 @@
     return x != null && x['@@functional/placeholder'] === true;
   };
 
-  /* istanbul ignore next */
-  var Any = function Any() {};
+  var formatters = {
+    '{}': R.identity,
+    '{ord}': R.nth(_, ['first', 'second', 'third']),
+    '{repr}': R.toString,
+    '{type}': function(type) { return /^function (\w*)/.exec(type)[1]; }
+  };
+
+  //  format :: (String, [*]) -> String
+  var format = function(template, values) {
+    var idx = -1;
+    return template.replace(/[{].*?[}]/g, function(match) {
+      return formatters[match](values[idx += 1]);
+    });
+  };
+
+  var a = {name: 'a'};
+  var b = {name: 'b'};
+  var c = {name: 'c'};
```

Comment:
> What are these variables?

They represent type variables, which place constraints on argument combinations.

In this case the arguments can be of any combination of types:

```javascript
//# K :: a -> b -> a
S.K = def('K', [a, b], function(x, y) {
  return x;
});
```

In this case the arguments can be of any type, but must both be of the same type:

```javascript
//# and :: a -> a -> a
S.and = def('and', [a, a], function(x, y) {
  return toBoolean(x) ? y : x;
});
```